stan/math/prim/mat/prob/multi_normal_rng.hpp

```diff
@@ -33,14 +46,21 @@ namespace stan {
       check_symmetric(function, "Covariance matrix", S);
       check_finite(function, "Location parameter", mu);
 
+      Eigen::LLT<Eigen::MatrixXd> llt_of_S = S.llt();
+      if (llt_of_S.info() != Eigen::Success) {
+        throw std::domain_error("multi_normal_rng:"
```

Comment:
No!  That causes the matrix to be factored twice.

- Bob

> On Sep 2, 2016, at 6:31 PM, Daniel Lee <notifications@github.com> wrote:
> 
> In stan/math/prim/mat/prob/multi_normal_rng.hpp:
> 
> > @@ -33,14 +46,21 @@ namespace stan {
> >        check_symmetric(function, "Covariance matrix", S);
> >        check_finite(function, "Location parameter", mu);
> >  
> > +      Eigen::LLT<Eigen::MatrixXd> llt_of_S = S.llt();
> > +      if (llt_of_S.info() != Eigen::Success) {
> > +        throw std::domain_error("multi_normal_rng:"
> 
> Want to just use check_pos_definite()? If so, you can replace lines 50-54 with:
> 
>   check_pos_definite("multi_normal_rng", "matrix argument", llt_of_S);
> 
> If you're going down path, could you change the error message on check_pos_definite? What you've written is so much clearer than "Cholesky decomposition of <> failed".
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>